src/cli/llm/wizard.rs

```diff
@@ -41,13 +46,32 @@ impl<Q, A> Wizard<Q, A> {
 
     pub async fn ask(&self, q: Q) -> Result<A>
     where
-        Q: TryInto<ChatRequest, Error = super::Error>,
-        A: TryFrom<ChatResponse, Error = super::Error>,
+        Q: TryInto<ChatRequest, Error = Error> + Clone,
+        A: TryFrom<ChatResponse, Error = Error>,
     {
-        let response = self
-            .client
-            .exec_chat(self.model.as_str(), q.try_into()?, None)
-            .await?;
-        A::try_from(response)
+        let retry_strategy = ExponentialBackoff::from_millis(1000).map(jitter).take(5);
+
+        Retry::spawn(retry_strategy, || async {
```

Comment:
Should use RetryIf over here. in the condition argument we should pass a closure that checks for the error type.